bin/compile

```diff
@@ -27,25 +39,29 @@ poetry export -f requirements.txt -o requirements.txt --without-hashes
 
 RUNTIME_FILE="runtime.txt"
 
-echo "-----> Export Python version from Poetry to Heroku $RUNTIME_FILE file"
+if [ ! "$DISABLE_POETRY_CREATE_RUNTIME_FILE" ] || [ "$DISABLE_POETRY_CREATE_RUNTIME_FILE" == "0" ]; then
```

Comment:
I would use an early return here instead of indenting the whole block. 

```bash
if [ "$DISABLE_POETRY_CREATE_RUNTIME_FILE" ] && [ "$DISABLE_POETRY_CREATE_RUNTIME_FILE" != "0" ]; then
  echo "-----> Skipping generation of $RUNTIME_FILE file from poetry.lock"
  exit 0
fi
```

Doesn't this work for you? Tested with:

```bash
export DISABLE_POETRY_CREATE_RUNTIME_FILE=1
export DISABLE_POETRY_CREATE_RUNTIME_FILE=true
export DISABLE_POETRY_CREATE_RUNTIME_FILE=0
export DISABLE_POETRY_CREATE_RUNTIME_FILE=
```

Reply:
> I would use an early return here instead of indenting the whole block.

I generally try to maintain a single exit-point from any call, including scripts. I've found it to be easier to manage and less prone to accidentally assuming the code will execute something placed near the end of a function/script by other contributors. Very happy to go back to the early return pattern though?

Reply:
Another idea: create a new function `export_heroku_runtime_file` and then in the "main" function  do something like `if [...]; then export_heroku_runtime_file(); fi`.

Reply:
I'd just go for an early return and be done with it. We can always refactor it into a function when/if it grows...

Reply:
I've reverted to the early return pattern.